### DISCUSSION TITLE
Not able to debug server side code in VS Code with Nuxt 3.5.1

### DISCUSSION THREAD

**Discussion by khaidoan** [nuxt/nuxt#21116] [Questions]
Hi,

My name is Khai.  I am using Nuxt 3.5.1, and I am trying to debug a composable on the server side using VS Code.  I am able set the breakpoint in VS Code, but as soon as I start the debugger, the breakpoint becomes disable and VS Code says "Unbound breakpoint".  I suspect that VS Code is not able to find the source map somehow. 

I've tried a number of things, but none worked.

Here is my launch.json file:

`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Nuxt remote debugging",
            "type": "node",
            "request": "attach",  
            "localRoot": "${workspaceFolder}",
            "remoteRoot": "/usr/src/app",
            "port": 9229,
            "address": "127.0.0.1",
            "sourceMaps": true,
            // .nuxt/dev/index.mjs.map
            // .nuxt/types/imports.d.ts
            "sourceMapPathOverrides": {
                "webpack:///": "${workspaceRoot}/",
                "webpack:///./~/": "${workspaceFolder}/node_modules/",
                "webpack://?:/": "${workspaceFolder}/*",
                "webpack:///./*": "${webRoot}/*"
            },
            "restart": true,
             //"trace": true
        }
    ]
}
`

And here is my nuxt.config.ts file:

`// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  ssr: true,

  routeRules: {
    '/vLoginRedirect': {ssr: false},
    '/vlogout': {ssr: false},
    '/**': {ssr: true},
    '/tasks/**': { swr: 5 },
    '/pages/tasks/**': { swr: 5 },
  },

  /*
  generate: {
    routes: ['/tasks/:id'],
  },
  */
  nitro: {
    /*
    sourceMap: true,
    routeRules: {
      '/tasks/**': { swr: 5 },
      '/pages/tasks/**': { swr: 5 },
    },
    */  
  },
  modules: [
    'dayjs-nuxt',
    '@formkit/nuxt',
    '@sfxcode/nuxt-primevue',
    // '@sfxcode/formkit-primevue'
  ],

  dayjs: {
    locales: ['en', 'fr'],
    plugins: ['relativeTime', 'utc', 'timezone'],
    defaultLocale: 'en',
    defaultTimezone: 'America/New_York',
  },

  primevue: {
    config: {
      ripple: true
    }
  },

  formkit: {
    configFile: '../formkit.config.ts'
  },

  css: [
    'primevue/resources/primevue.css',
    'primevue/resources/themes/saga-blue/theme.css',
    'primeicons/primeicons.css',
    '@sfxcode/formkit-primevue/dist/sass/formkit-prime-inputs.scss',
    '@sfxcode/formkit-primevue/dist/sass/formkit-primevue.scss'
  ],

  build: {
    transpile: ['primevue'],
    analyze: true
  },

  app: {
      head: {
        meta: [
          { name: 'charset', content: 'utf-8' },
          { name: 'viewport', content: 'width=device-width, initial-scale=1' }
        ],
        script: [
          { src: '/vstatic/bootstrap-5.3/bootstrap.bundle.min.js' },
          { src: 'https://kit.fontawesome.com/b432aaf6cf.js' }
        ],
        link: [
          // <link rel="stylesheet" href="https://myawesome-lib.css">
          { rel: 'stylesheet', href: '/vstatic/bootstrap-5.3/bootstrap.min.css' },
          { rel: 'stylesheet', href: '/vstatic/css/styles.css' }
        ]
        /*,
        noscript: [
          // <noscript>JavaScript is required</noscript>
          { children: 'JavaScript is required' }
        ]*/
      }
    },

  runtimeConfig: {
    api_key: process.env.API_KEY,
    private_api_gateway: process.env.PRIVATE_API_GATEWAY,
    public: {
      public_api_gateway: process.env.PUBLIC_API_GATEWAY
    }
  },

  vite: {
    build: {
      sourcemap: 'inline'
    },
    server: {
      hmr: {
        protocol: "wss",
        path: "hmr/",
      }
    }
  },

  vue: {
    compilerOptions: {
      sourceMap: true,
    }
  },

  hooks: {
    'vite:extendConfig': function (clientConfig, { isClient, isServer }) {
      if (clientConfig.mode === 'development') {
        if (isClient)
          clientConfig.build.sourcemap = true // defaults to true in development
        if (isServer)
          clientConfig.build.sourcemap = 'inline' // defaults to true in development
      }
    },
  },

  devtools: {
    enabled: false
  },
  
  sourcemap: {
    "server": true,
    "client": true
  },
  debug: true
})
`

I've looked inside the .nuxt/dist/server folder, and I do not see any .map file.  I spent hours on this but not able to debug with VS Code.  I am new to Nuxt.  Are you facing the same issue?  Can you tell me where and how to find the source map files?

Thank you!
Khai

**Comment by Clive-Ward-Cron** [comment]
You may have read this before, but here is the information from the Nuxt documentation: [https://nuxt.com/docs/guide/going-further/debugging#example-vs-code-debug-configuration](url)

I've also had the same issue with attaching to the node process that is running Nuxt. I haven't found a way to bind the breakpoints by attaching, **BUT** I have been able to bind the node breakpoints by running the dev process using the VS code debugger. It's labelled "server: nuxt" in the documentation link I provided.

**NOTE:** You may need to open VS Code from the terminal by using `code .` in the directory you're working from. If I don't do that it still won't bind.

If you're having issues attaching to your browser, make sure you're running it with the debug port configured. I set it on the target property of my Chrome shortcut in my taskbar.

![image](https://github.com/nuxt/nuxt/assets/23231126/ad1fd6ea-176b-481f-817c-98c028421ff9)
